### DISCUSSION TITLE
Why LZ4 1.10 multi-threading compression doesn't support in-memory data compression

### DISCUSSION THREAD

**Discussion by shichumy** [lz4/lz4#1528] [Q&A]
Hi @Cyan4973,

In the release of LZ4 version 1.10, the multi-threading compression feature seems to only support file compression and does not extend to compressing in-memory data.  I would like to know why in-memory multi-threading compression is not supported. Do you have plans to support it in the future? If so, when would the multi-threading compression support in-memory data compression?

I look forward to your response. Thank you!

**Comment by Cyan4973** [comment]
It's not obvious.

`liblz4` has also a promise of small size, both binary and source code, and low dependency surface.
Throwing a multi-threaded variant into the mix has the potential to break that promise.

Now, I'm not saying it's impossible, but it shall be made with care, and attention to above properties.
It's likely doable, but not simple, and will require time, which I can ill afford today.

So maybe an item for some future release.
